irohad/ametsuchi/impl/rocksdb_command_executor.cpp

```diff
@@ -0,0 +1,293 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ametsuchi/impl/rocksdb_command_executor.hpp"
+
+#include <fmt/core.h>
+#include <rocksdb/utilities/transaction.h>
+#include <boost/algorithm/string.hpp>
+#include <boost/variant/apply_visitor.hpp>
+#include "ametsuchi/impl/executor_common.hpp"
+#include "ametsuchi/setting_query.hpp"
+#include "ametsuchi/vm_caller.hpp"
+#include "interfaces/commands/add_asset_quantity.hpp"
+#include "interfaces/commands/add_peer.hpp"
+#include "interfaces/commands/add_signatory.hpp"
+#include "interfaces/commands/append_role.hpp"
+#include "interfaces/commands/call_engine.hpp"
+#include "interfaces/commands/command.hpp"
+#include "interfaces/commands/compare_and_set_account_detail.hpp"
+#include "interfaces/commands/create_account.hpp"
+#include "interfaces/commands/create_asset.hpp"
+#include "interfaces/commands/create_domain.hpp"
+#include "interfaces/commands/create_role.hpp"
+#include "interfaces/commands/detach_role.hpp"
+#include "interfaces/commands/grant_permission.hpp"
+#include "interfaces/commands/remove_peer.hpp"
+#include "interfaces/commands/remove_signatory.hpp"
+#include "interfaces/commands/revoke_permission.hpp"
+#include "interfaces/commands/set_account_detail.hpp"
+#include "interfaces/commands/set_quorum.hpp"
+#include "interfaces/commands/set_setting_value.hpp"
+#include "interfaces/commands/subtract_asset_quantity.hpp"
+#include "interfaces/commands/transfer_asset.hpp"
+
+using namespace iroha;
+using namespace iroha::ametsuchi;
+
+using shared_model::interface::permissions::Grantable;
+using shared_model::interface::permissions::Role;
+
+using shared_model::interface::GrantablePermissionSet;
+using shared_model::interface::RolePermissionSet;
+
+RocksDbCommandExecutor::RocksDbCommandExecutor(
+    std::shared_ptr<RocksDBPort> db_port,
+    std::shared_ptr<shared_model::interface::PermissionToString> perm_converter,
+    std::optional<std::reference_wrapper<const VmCaller>> vm_caller)
+    : db_port_(std::move(db_port)),
+      perm_converter_{std::move(perm_converter)},
+      vm_caller_{std::move(vm_caller)} {
+  db_port_->prepareTransaction(*db_context_);
+}
+
+RocksDbCommandExecutor::~RocksDbCommandExecutor() = default;
+
+CommandResult RocksDbCommandExecutor::execute(
+    const shared_model::interface::Command &cmd,
+    const shared_model::interface::types::AccountIdType &creator_account_id,
+    const std::string &tx_hash,
+    shared_model::interface::types::CommandIndexType cmd_index,
+    bool do_validation) {
+  return boost::apply_visitor(
+      [this, &creator_account_id, &tx_hash, cmd_index, do_validation](
+          const auto &command) -> CommandResult {
```

Comment:
At this point we know what command to execute. Function execute() could be a template for CommandType.

Please remove overload by operator(). Make code simpler.